packages/primero/src/Adaptor.js

```diff
@@ -176,9 +176,19 @@ function cleanupState(state) {
  *   remote: true,
  *   case_id: "6aeaa66a-5a92-4ff5-bf7a-e59cde07eaaz",
  * });
+ * @example <caption>Get case from Primero with referrals</caption>
+ * getCases(
+ *   {
+ *     remote: true,
+ *   },
+ *   {
+ *     withReferrals: true,
+ *   }
+ * );
  * @function
- * @param {object} query - an object with a query param at minimum, option to getReferrals
- * @param {object} options - (Optional) an object with a getReferrals key to fetch referrals
+ * @param {object} query - an object with a query param at minimum, option to getCases
+ * @param {object} [options] - (Optional) an object with a withReferrals key to fetch referrals
```

Comment:
I'm not really happy about the clarity of these docs. Would a typedef help?

Biggest thing for me though: we should mention that referrals may take a while. It's probably ok to mention that each referral generates a new request (if we word it well)

Reply:
It's very tricky defining the params for `query` object, because so far the only reference i have is what is on the examples for `getCases()`. Maybe i should just document those for now ?

Reply:
Ok, first off, the extra documentation we've added really doesn't shed much light on what the query options are.

For example: `[query.remote] - A remote option to getCases`

I mean, Ok at least we're declaring that there is a `remote` option now. But we're not explaining what this means.

If  we're going to take the time to add documentation, it has to actually be meaningful, you know?

Also this sentence doesn't make a lot of sense:
```
 @param {object} [query] - (Optional) an object with a query param at minimum, option to getCases
```
I think it's saying that `query` is required, but I don't even think that's true.

Interestingly, the main `getCases` docs does shed light on some of this stuff:

>Note that in many implementations, the remote attribute should be set to true to ensure that only cases marked for remote access will be retrieved. You can specify a case_id value to fetch a unique case and a query string to filter result.

So that's a start. Linking to the right part of Primero docs might also be useful.

Then we've got the `withReferrals` problem, which really shouldn't be in its own object. But I don't want to change that now.

I would simply do this:
```
@param {object} [options] - Additional options
@param {boolean} [options.withReferrals] - Set to true to fetch referrals. This will generate an extra request for each case and may take some time to process.
```

Reply:
The documentation for this is bugging me as well and mainly because i can't find a good Primero API documentation, here is my suggestion let's remove the query params documentation for now and [plan for primero epic](https://github.com/OpenFn/adaptors/issues/680)

Reply:
Those docs look pretty clear to me, and I think our current documentation is completely wrong.

Leave this with me - I'll get it sorted today.